### PR TITLE
Fix j385 nat ip change - lan ip change resulting in NATed port change should be fixed, we correctly compare IPs+port in map now

### PR DESCRIPTION
**Pull request by rfreemobile**

### CODE REVIEW

File: src/tunserver.cpp [hunk 1]
```diff
@@ -845,7 +845,7 @@ bool c_tunserver::route_tun_data_to_its_destination_top(t_route_method method,
 
 c_peering & c_tunserver::find_peer_by_sender_peering_addr( c_ip46_addr ip ) const {
 	std::lock_guard<std::mutex> lg(m_peer_mutex);
-	for(auto & v : m_peer) { if ((v.second->get_pip() == ip) && (v.second->get_pip().get_assign_port() == ip.get_assign_port())) return * v.second.get(); }
+	for(auto & v : m_peer) { if ((v.second->get_pip() == ip) && (v.second->get_pip().get_assigned_port() == ip.get_assigned_port())) return * v.second.get(); }
```
Comment (tigusoft) [inline src/tunserver.cpp]: now this additional port check is not needed.
@rfreemobile TODO

File: src/test/c_ip46_addr.cpp [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+			if (i<j)  EXPECT_TRUE(   tab.at(i) <  tab.at(j) );
+			if (i>j)  EXPECT_TRUE(!( tab.at(i) <  tab.at(j) ));
+
+			if (i==j) EXPECT_TRUE(   tab.at(i) == tab.at(j) );
+			if (i!=j) EXPECT_TRUE(!( tab.at(i) == tab.at(j) ));
+		}
+	}
+
+}
+
```
Comment (tigusoft) [inline src/test/c_ip46_addr.cpp]: Still needs more UT for corner cases of comparing ipv4 with ipv6, and with none.
TODO: Document what is allowed and what throws (in class) and test.
